managed/src/test/java/com/yugabyte/yw/models/InstanceTypeTest.java

```diff
@@ -90,7 +94,7 @@ public void testFindByProvider() {
     instanceType.setActive(false);
     instanceType.save();
     InstanceType.upsert(newProvider.code, "bar", 2, 10.0, defaultDetails);
```

Comment:
This is incorrect. Not sure what is your confusion.
Here is what you have to do in great detail:
1. Revert to using mock.Config 
2. See what is in defaultDetails  - line 44 populates it. But we do not want to test that code path. To excercise our code path the defaultDetails has to be empty. Thats when the populateEmpty will do something and try to lookup config values.
3. Like line 88 upsert another instancetype  such that its details have volumeList empty.
4. Now the test will fail complaining that you have not mocked config.getInt or whatever you use. 
5. Add mock setup. For example volume count config return 3 and size returns 100GB.
6.  assert that the volume list for this instance as returned by findPyProvider is size 3 and each one is 100GB.

Hope it helps.

Reply:
Had to refactor the code lil bit. Forgot to apply @RunWith(MockitoJUnitRunner.class) on top to use mock. Now its working fine. I really learnt a lot while working on this change.  Please let me know if you still thinks, some changes are required. Thanks for all your suggestions:)

Reply:
💯  for right attitude!